pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -880,7 +883,7 @@ func (adc *attachDetachController) GetServiceAccountTokenFunc() func(_, _ string
 
 func (adc *attachDetachController) DeleteServiceAccountTokenFunc() func(types.UID) {
 	return func(types.UID) {
-		klog.ErrorS(nil, "DeleteServiceAccountToken unsupported in attachDetachController")
+		adc.logger.Error(nil, "DeleteServiceAccountToken unsupported in attachDetachController")
```

Comment:
To confirm my understanding, should I modify this function into following style?
```
func (adc *attachDetachController) DeleteServiceAccountTokenFunc() func(logger klog.Logger, types.UID) {
      return func(logger klog.Logger, types.UID) {
           logger.Error(....)
     }
}
```
@pohly

Reply:
Correct.

Reply:
That probably would have consequences elsewhere. We can also do this:
```patch
diff --git a/pkg/controller/volume/attachdetach/attach_detach_controller.go b/pkg/controller/volume/attachdetach/attach_detach_controller.go
index 643993d2e7d..8c86a3076a1 100644
--- a/pkg/controller/volume/attachdetach/attach_detach_controller.go
+++ b/pkg/controller/volume/attachdetach/attach_detach_controller.go
@@ -137,7 +137,6 @@ func NewAttachDetachController(
                cloud:               cloud,
                pvcQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "pvcs"),
                filteredDialOptions: filteredDialOptions,
-               logger:              logger,
        }
 
        adc.csiNodeLister = csiNodeInformer.Lister()
@@ -883,7 +882,7 @@ func (adc *attachDetachController) GetServiceAccountTokenFunc() func(_, _ string
 
 func (adc *attachDetachController) DeleteServiceAccountTokenFunc() func(types.UID) {
        return func(types.UID) {
-               adc.logger.Error(nil, "DeleteServiceAccountToken unsupported in attachDetachController")
+               klog.ErrorS(nil, "DeleteServiceAccountToken unsupported in attachDetachController") //nolint:logcheck // should never be reached
        }
 }
 
```

Alternatively, we can panic here instead of logging an error. It shouldn't matter when the code never gets reached (and it shouldn't get reached, otherwise production logs would contain this error) and if it gets reached, a panic is better than continuing and hoping that someone notices.

I've asked about this in https://kubernetes.slack.com/archives/C5P3FE08M/p1686744938646189

Let's use the nolint comment for now because it's closer to the original code.

Reply:
So you mean we should change like following? @pohly
```

 func (adc *attachDetachController) DeleteServiceAccountTokenFunc() func(types.UID) {
        return func(types.UID) {
               // nolint:logcheck
               klog.ErrorS(nil, "DeleteServiceAccountToken unsupported in attachDetachController") //nolint:logcheck // should never be reached
        }
 }
```

Reply:
add nolint